### PR TITLE
Test cases

### PR DESCRIPTION
**Pull request by theRealRizeo**
Replaces PHPUnit tests with codeception.

Requires GitHub actions enabled https://github.com/marketplace/actions/codeception-action

Instructions for local setup can be found at the `TESTS.md` file in the plugin root

### CODE REVIEW

**Comment by stephywells** [comment]
@theRealRizeo it looks like this isn't passing here in Github. Is there a way to get it passing?

**Comment by theRealRizeo** [comment]
@stephywells github actions in the plugin are not made to run these cedeception tests hence the failures

**Comment by stephywells** [comment]
@theRealRizeo what can we do to get them working? We need this automated testing to run in Github too.

**Comment by theRealRizeo** [comment]
@stephywells I have added the actions that need to be enabled in the description, there will need to be some configurations as this does not run the same as default phpunit.  Also there is an error being thrown on composer install

```
[RuntimeException]                                                                                            
  Failed to clone 
https://github.com/hautelook/phpass.git via https, ssh protocols, aborting.                   

  - 
https://github.com/hautelook/phpass.git
    Cloning into bare repository '/home/runner/.cache/composer/vcs/https---github.com-hautelook-phpass.git'...  
    remote: Invalid username or password.                                                                       
    fatal: Authentication failed for '
https://github.com/hautelook/phpass.git/'                                 

  - git@github.com:hautelook/phpass.git                                                                         
    Cloning into bare repository '/home/runner/.cache/composer/vcs/https---github.com-hautelook-phpass.git'...  
    Warning: Permanently added the RSA host key for IP address '140.82.113.3' to the list of known hosts.       
    git@github.com: Permission denied (publickey).                                                              
    fatal: Could not read from remote repository.                                                               

    Please make sure you have the correct access rights                                                         
    and the repository exists.                                                                                  
```

And the webdriver also needs to be configured

**Comment by stephywells** [comment]
@theRealRizeo can you work on the GitHub action to get this working here too?

**Comment by theRealRizeo** [comment]
@stephywells got it working. 1 test failed due to errors fetching mysqli so I have skipped it for now, its an issue with getting the payments for the listing in one test. The other tests work fine

File: tests/_support/Helper/Unit.php [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?php
+namespace Helper;
+
+// here you can define custom actions
+// all public methods declared in helper class will be available in $I
+
+class Unit extends \Codeception\Module {
```
Comment (stephywells) [inline tests/_support/Helper/Unit.php:7]: Awesome job on this! 

I'm not familiar with Code epitome at all yet. Can you tell me if these empty classes are required?

Comment (theRealRizeo) [reply tests/_support/Helper/Unit.php:7]: Yes, the files in the _support directory are required https://codeception.com/docs/06-ModulesAndHelpers under `Helpers`

Comment (theRealRizeo) [reply tests/_support/Helper/Unit.php:7]: The files are actually generated from the bootstrap method https://codeception.com/quickstart

**Review by stephywells — APPROVED** [review]

**Pull request merged**